### PR TITLE
fix(shorebird_cli): `shorebird init` iOS flavor detection on non-macOS

### PR DESCRIPTION
**Pull request by felangel**
## Description

- fix(shorebird_cli): `shorebird init` iOS flavor detection on non-macOS (part of #910)

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [X] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/init_command.dart [hunk 1]
```diff
@@ -172,11 +174,35 @@ For more information about Shorebird, visit ${link(uri: Uri.parse('https://shore
...
+    if (platform.isMacOS) {
+      try {
+        final info = await xcodeBuild.list(projectPath);
+        return info.schemes.whereNot((element) => element == 'Runner').toSet();
+      } on MissingIOSProjectException {
+        return null;
+      }
+    } else {
+      // When running on a non-macOS platform, we can't use xcodebuild to
+      // detect flavors so we fallback to looking for schemes in xcschemes.
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/commands/init_command.dart]: Worth noting here that this seems to be exactly the behavior of xcodebuild

File: packages/shorebird_cli/test/src/commands/init_command_test.dart [hunk 2]
```diff
@@ -247,6 +255,106 @@ If you want to reinitialize Shorebird, please run "shorebird init --force".''',
       expect(exitCode, ExitCode.software.code);
     });
 
+    test('throws software error when unable to detect schemes (non-macos)',
+        () async {
+      when(() => platform.isMacOS).thenReturn(false);
```
Comment (bryanoltman) [inline packages/shorebird_cli/test/src/commands/init_command_test.dart]: This is stylistic and completely optional, but I've found that this kind of test becomes slightly more readable when put in groups:

```
group('on macOS', () {
  setUp(() {
    when(() => platform.isMacOS).thenReturn(true);
  });

  test('throws software error when unable to detect schemes', ...)
});

group('on a non-mac OS', () {
  setUp(() {
    when(() => platform.isMacOS).thenReturn(false);
  });

  test('throws software error when unable to detect schemes', ...)
});
```

**Review by bryanoltman — APPROVED** [review]

File: packages/shorebird_cli/test/src/commands/init_command_test.dart (hunk 2 above)
Comment (felangel) [reply packages/shorebird_cli/test/src/commands/init_command_test.dart]: yup good call will reorganize.

File: packages/shorebird_cli/lib/src/commands/init_command.dart (hunk 1 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/commands/init_command.dart]: done ✅

**Pull request merged**